### ISSUE TITLE
Failing in MSVC with c++17 syntax

### ISSUE THREAD

**Issue by AntonioCS** [hanickadot/compile-time-regular-expressions#80]
Hey,

I had some code working in GCC (using CE) but when I switched to MSVC 19 everything failed.
Here is the code: 
```
std::vector<std::pair<std::string_view, int>> match_all(std::string_view input) noexcept {
    std::vector<std::pair<std::string_view, int>> matches{};
    using namespace ctre::literals;

    auto pattern = ctll::fixed_string{ "(\\{\\{|\\{%|\\{#(\\-|~)?)" };
    
    //for (auto match : ctre::range<"(\\{\\{|\\{%|\\{#(\\-|~)?)">(input)) { //works in gcc
    for (auto match : ctre::range<pattern>(input)) { //fails
    //for (auto match : "(\\{\\{|\\{%|\\{#(\\-|~)?)"_ctre.range(input)) { //also fails
        auto res = match.get<1>().to_view();
        matches.push_back({res, res.data() - input.data()}); 
    }

    return matches;
}
```
Full example on CE: https://gcc.godbolt.org/z/4bXy_M

I believe I have followed the examples correctly but everything fails and also I couldn't get the `_ctre` postfix to work, but that just might be my noobish level in c++

**Comment by hanickadot** [comment]
Not a bug, MSVC doesn't support cNTTP or UDL syntax as described in [readme](https://github.com/hanickadot/compile-time-regular-expressions#supported-compilers). You must use C++17 syntax, so your pattern must be in `static constexpr` variable with linkage.

TL;DR: your `auto pattern` must be `static constexpr auto pattern` and outside of function.

**Comment by AntonioCS** [comment]
Apologies for my noobishness!
